### DISCUSSION TITLE
How to add Cache Control to the response header of  “actix_files” ？

### DISCUSSION THREAD

**Discussion by 154286** [actix/actix-web#3260] [Q&A] [answered]
```  
 App::new()
            .app_data(json_config)
            .service(actix_files::Files::new("/images", &CONFIG.images.dir).use_last_modified(true))
            .wrap(middleware::DefaultHeaders::new().add(("Cache-Control", "no-cache")))
```

This can achieve my goal, but it will result in all interfaces carrying this header
How can I only add Cache Control to the/images route without affecting other interfaces？

Methods like 
```
 .service(actix_files::Files::new("/images", &CONFIG.images.dir).use_last_modified(true).wrap(middleware::DefaultHeaders::new().add(("Cache-Control", "no-cache"))))
```

are not feasible

**Comment by robjtede** [comment, marked as answer]
Here's what I'd recommend:

```rust
.service(
    actix_web::web::scope("/images")
        .default_service(
            actix_files::Files::new("", &CONFIG.images.dir).use_last_modified(true),
        )
        .wrap(DefaultHeaders::new()...),
)
```

It does feel natural to want to wrap the files service like this though, I should have a think about making it easier.

**Reply by 154286** [reply]
Thank you for your suggestion. It can indeed meet my needs

**Comment by Samielakkad** [comment]
`DefaultHeaders` applies to the service it wraps. Since `Files::new(...)` itself does not expose `.wrap(...)` the same way a `Scope` does, wrap a scope and put the files service inside it.

Example:

```rust
use actix_web::{middleware::DefaultHeaders, web};

App::new()
    .service(
        web::scope("/images")
            .wrap(
                DefaultHeaders::new()
                    .add(("Cache-Control", "no-cache")),
            )
            .service(
                actix_files::Files::new("", &CONFIG.images.dir)
                    .use_last_modified(true),
            ),
    )
```

The important part is that the mount path moves to the scope:

```rust
web::scope("/images")
Files::new("", dir)
```

instead of:

```rust
Files::new("/images", dir)
```

That way only responses under `/images` get the cache header, and the rest of the app is unaffected.
